src/source.rs

```diff
@@ -538,6 +548,70 @@ where
+                        codec_override,
+                        stream,
+                    }) => {
+                        for (data, meta_data) in batch_data {
+                            origin_uri.maybe_set_uid(self.uid);
+                            let mut ingest_ns = nanotime();
+                            let mut error = false;
+                            let original_id = self.id;
+                            let results = self
+                                .make_event_data(
+                                    stream,
+                                    &mut ingest_ns,
+                                    codec_override.clone(),
+                                    data,
+                                    meta_data.map(StaticValue),
+                                )
+                                .await;
+                            if results.is_empty() {
+                                self.source.on_empty_event(original_id, stream).await?;
+                            }
+
+                            for result in results {
+                                let (port, data) = match result {
+                                    Ok(d) => (OUT, d),
+                                    Err(e) => {
+                                        error!(
+                                            "[Source::{}] Error decoding event data: {}",
+                                            self.source_id, e
+                                        );
+                                        let mut error_meta = Object::with_capacity(1);
+                                        error_meta
+                                            .insert_nocheck("error".into(), e.to_string().into());
+
+                                        let mut error_data = Object::with_capacity(3);
+                                        error_data
+                                            .insert_nocheck("error".into(), e.to_string().into());
+                                        error_data
+                                            .insert_nocheck("event_id".into(), original_id.into());
+                                        error_data.insert_nocheck(
+                                            "source_id".into(),
+                                            self.source_id.to_string().into(),
+                                        );
+                                        (
+                                            ERR,
+                                            (Value::from(error_data), Value::from(error_meta))
+                                                .into(),
+                                        )
+                                    }
+                                };
+                                error |= self
+                                    .transmit_event(data, ingest_ns, origin_uri.clone(), port)
+                                    .await;
+                            }
```

Comment:
This would wonderfully work as a ownfunction with the code below also calling it

Reply:
To be fixed, working on it